src/cases/containers/case-search/case-search.component.ts

```diff
@@ -1,31 +1,160 @@
+    this.metadataFields$ = this.store.pipe(select(fromCasesFeature.searchFilterMetadataFields));
+
+    this.filterSubscription = combineLatest([
+      this.jurisdiction$,
+      this.caseType$,
+      this.caseState$,
+      this.metadataFields$
+    ]).subscribe(result => {
+      this.jurisdiction = {
+        ...result[0]
+      };
+      this.caseType = {
+        ...result[1]
+      };
+      this.caseState = {
+        ...result[2]
+      };
+      this.metadataFields = {
+        ...result[3]
+      };
+    });
+
+    this.resultSubscription = this.resultView$.subscribe(resultView => {
+
+      this.paginationMetadata = new PaginationMetadata();
+      const resultViewResultsLength = resultView.results ? resultView.results.length : 0;
+      this.paginationMetadata.total_results_count = resultViewResultsLength;
+      this.paginationMetadata.total_pages_count = Math.ceil(resultViewResultsLength / this.paginationSize);
+      this.resultsArr = resultView.results;
+      this.resultView = {
+        ...resultView,
+        columns: resultView.columns ? resultView.columns : [],
+        results: resultView.results ? resultView.results.map(item => {
+          return {
+            ...item,
+            hydrated_case_fields: null
+          };
+        }).slice(0, this.paginationSize) : [],
+        hasDrafts: resultView.hasDrafts ? resultView.hasDrafts : () => false
+      };
+    });
+
+    this.checkLSAndTrigger();
+  }
+
+  checkLSAndTrigger() {
+
+    const formGroupFromLS = JSON.parse(localStorage.getItem('search-form-group-value'));
+    const jurisdictionFromLS = JSON.parse(localStorage.getItem('search-jurisdiction'));
+    const caseTypeGroupFromLS = JSON.parse(localStorage.getItem('search-caseType'));
+    const metadataFieldsGroupFromLS = JSON.parse(localStorage.getItem('search-metadata-fields'));
+
+    if (formGroupFromLS && jurisdictionFromLS && caseTypeGroupFromLS &&   metadataFieldsGroupFromLS) {
+      const event = {
+        selected: {
+          jurisdiction: jurisdictionFromLS,
+          caseType: caseTypeGroupFromLS,
+          metadataFields: metadataFieldsGroupFromLS,
+          formGroup: {
+            value: formGroupFromLS
+          }
+        }
+      };
+
+      this.store.dispatch(new fromCasesFeature.ApplySearchFilter(event));
+    }
+  }
+
+  applyChangePage(event) {
+    if (this.resultsArr.length > 0) {
+      const startingPoint = (event.selected.page - 1) * this.paginationSize;
+      const endingPoint = startingPoint + this.paginationSize;
+      const newArr = this.resultsArr.map(item => {
+        return {
+          ...item,
+          hydrated_case_fields: null
+        };
+      }).slice(startingPoint, endingPoint);
+
+      this.resultView = {
+        ...this.resultView,
+        results: newArr,
+        hasDrafts: this.resultView.hasDrafts
+      };
+    }
+  }
+
+  ngOnDestroy() {
+    this.filterSubscription.unsubscribe();
```

Comment:
Please check if subscription is not null/undefined and then unsubscribe
if(this.filterSubscription){
this.filterSubscription.unsubscribe();
}

Reply:
Good call. Done.